pandas/__init__.py

```diff
@@ -202,8 +202,40 @@
         FutureWarning,
         stacklevel=2,
     )
-# Don't allow users to use pandas.os or pandas.warnings
-del os, warnings
+
+# DeprecationWarning for missing pyarrow
+from pandas.compat.pyarrow import pa_version_under10p1
+
+if pa_version_under10p1:
+    # Check if old pyarrow is installed
+    from pandas.compat._optional import VERSIONS
+
+    try:
+        import pyarrow
+
+        pa_msg = (
+            f"was too old on your system - pyarrow {VERSIONS['pyarrow']} "
+            "is the current minimum supported version as of this release."
+        )
+        del pyarrow
+    except ImportError:
+        pa_msg = "was not found to be installed on your system."
+
+    warnings.warn(
+        f"""
+Pyarrow will become a required dependency of pandas in the next major release of pandas,
+(to allow more performant data types and better interoperability with other libraries)
```

Comment:
I would maybe explicitly mention the new default string dtype, since that's still the only one that is used by default that requires pyarrow

Reply:
Updated.